mongodb/terraform-provider-mongodbatlas
# feat: Migrate DataSource: mongodbatlas_project_ip_access_list to Terraform Plugin Framework

## Description

Ticket: [INTMDB-975](https://jira.mongodb.org/browse/INTMDB-975)

This PR migrates `mongodbatlas_project_ip_access_list` to the new framework

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)

## Further comments

### Testing
In addition to the added acceptance tests I tested the data source locally in the following scenario:

- Create IP access list entry in 1.11.0 + data source, upgrade to local build, run `tf plan` and verify plan was empty

<details>
<summary> tf plan </summary> 

```bash
data.mongodbatlas_project_ip_access_list.test: Read complete after 0s [id=64df3738bbba4d088e1cbbdf-1.2.3.4]
data.mongodbatlas_database_user.test: Read complete after 0s [id=YXV0aF9kYXRhYmFzZV9uYW1l:YWRtaW4=-cHJvamVjdF9pZA==:NjRkZjM3MzhiYmJhNGQwODhlMWNiYmRm-dXNlcm5hbWU=:dGVzdA==]
2023-08-18T10:46:27.816+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-18T10:46:27.818+0100 [DEBUG] provider: plugin process exited: path=../bin/terraform-provider-mongodbatlas pid=96650
2023-08-18T10:46:27.818+0100 [DEBUG] provider: plugin exited
2023-08-18T10:46:27.820+0100 [DEBUG] no planned changes, skipping apply graph check
2023-08-18T10:46:27.820+0100 [INFO]  backend/local: plan operation completed

No changes. Your infrastructure matches the configuration.
```

</details>


- Create IP access list entry + data source in local build 

<details>
<summary> tf state</summary> 

```terraform
# data.mongodbatlas_project_ip_access_list.test-ip:
data "mongodbatlas_project_ip_access_list" "test-ip" {
    cidr_block = "1.2.3.4/32"
    comment    = "cidr block for tf acc testing"
    id         = "64df3738bbba4d088e1cbbdf-1.2.3.4"
    ip_address = "1.2.3.4"
    project_id = "64df3738bbba4d088e1cbbdf"
}

# data.mongodbatlas_project_ip_access_list.test:
data "mongodbatlas_project_ip_access_list" "test" {
    cidr_block = "1.2.3.4/32"
    comment    = "cidr block for tf acc testing"
    id         = "64df3738bbba4d088e1cbbdf-1.2.3.4"
    ip_address = "1.2.3.4"
    project_id = "64df3738bbba4d088e1cbbdf"
}
```
</details>
